### PR TITLE
Create container runtime smoke test

### PR DESCRIPTION
**Pull request by pdostal**
As we test multiple container runtimes and run test in more and more environments I want to create simple and unversal smoke test.  

- Related ticket: [poo#105912](https://progress.opensuse.org/issues/105912)
- Verification run: [pdostal-server](http://pdostal-server.suse.cz/tests/13363#step/containerd_crictl/75)

### CODE REVIEW

**Comment by pdostal** [comment]
I am stucked with [crictl run](https://github.com/kubernetes-sigs/cri-tools/blob/master/docs/crictl.md#create-and-start-a-container-within-one-command) which needs pod config file, container config file and is failing because of `cni plugin not initialized` (see in [openQA](http://pdostal-server.suse.cz/tests/13363#step/containerd_crictl/75)).

**Comment by pdostal** [comment]
I'm removing this from `runtime_smoke_tests()`:
```perl
assert_script_run("curl -o pod-config.json " . data_url("containers/pod-config.json"));
assert_script_run("curl -o container-config.json " . data_url("containers/container-config.json"));
script_run("crictl run container-config.json pod-config.json");
```
The `pod-config.json` file:
```json
{
    "metadata": {
        "name": "nginx-sandbox",
        "namespace": "default",
        "attempt": 1,
        "uid": "hdishd83djaidwnduwk28bcsb"
    },
    "log_directory": "/tmp",
    "linux": {
    }
}
```
The `container-config.json` file:
```json
{
  "metadata": {
      "name": "busybox"
  },
  "image":{
      "image": "busybox"
  },
  "command": [
      "top"
  ],
  "log_path":"busybox.0.log",
  "linux": {
  }
}
```

**Comment by pdostal** [comment]
I'd also like to enable this smoke test in migration scenario (see #14036) cc @lemon-suse.

**Comment by lemon-suse** [comment]
> I'd also like to enable this smoke test in migration scenario (see #14036) cc @lemon-suse.

Let's wait the VR, http://openqa.nue.suse.com/tests/8206162#live

File: lib/services/docker.pm [hunk 1]
```diff
@@ -36,14 +36,11 @@ sub full_docker_check {
 
     check_service();
 
-    # Test the connectivity of Docker containers
-    $engine->check_containers_connectivity();
+    # Test the basic functionality
+    runtime_smoke_tests(runtime => $engine);
```
Comment (lemon-suse) [inline lib/services/docker.pm:40]: Need include  runtime_smoke_tests before run it.

Comment (pdostal) [reply lib/services/docker.pm:39]: I'm sorry. It's fixed now. Can you rerun the VR?

Comment (lemon-suse) [reply lib/services/docker.pm:39]: Ok, re-run it, http://openqa.nue.suse.com/tests/8206368#live

**Comment by lemon-suse** [comment]
The new VR passed, http://openqa.nue.suse.com/tests/8206368#step/check_upgraded_service/194

**Review by grisu48 — APPROVED** [review]
This is awesome! :rocket: When can we merge this?

File: lib/containers/utils.pm [hunk 2]
```diff
@@ -78,6 +79,48 @@ sub registry_url {
...
+    # if we miss $image the test will fail later
+    assert_script_run("$runtime image ls");
+
+    # crictl is not implemented as it needs a lot of additional settings just to run container
+    if ($runtime !~ /crictl/) {
+        # Run container in foreground
+        assert_script_run("$runtime run -it --rm $image echo 'Hello'");
+
+        # Run container in background - it can take few seconds
+        assert_script_run("$runtime run -d --name 'sleeper' $image sleep 999");
```
Comment (jlausuch) [inline lib/containers/utils.pm:107]: you can also do `"$runtime run -d --name 'sleeper' $image sh"` and it will hang forever

Comment (pdostal) [reply lib/containers/utils.pm:107]: I was not sure about this. We can improve it the next time.

File: lib/containers/utils.pm [hunk 3]
```diff
@@ -78,6 +79,48 @@ sub registry_url {
...
+    # Pull image from registry
+    if ($runtime =~ /nerdctl/) {
+        assert_script_run("$runtime image pull --insecure-registry $image");
+    } else {
+        assert_script_run("$runtime pull $image");
+    }
+
+    # List locally available images
+    # if we miss $image the test will fail later
+    assert_script_run("$runtime image ls");
```
[the remark this answers is not in the document]
Comment (pdostal) [reply lib/containers/utils.pm:99]: Well not really, you cannot simply ` | grep $image`.

But this in my opinion does not matter as the next command will anyways fail if the image isn't available.

I really tried to keep this as simple as possible without any parsing and output analyzing.

**Pull request merged**